### PR TITLE
Ability to add a theme during installation

### PR DESCRIPTION
**Pull request by Shine-neko**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Ability to add a theme during installation |
| Type? | new feature |
| Category? | IN |
| BC breaks? | no |
| Deprecations? | no |
| How to test it ? | run : php install-dev/index_cli.php --domain=example.com --db_server=sql.example.com --db_name=prestashop --db_user=root --db_password=azerty123 --theme=StarterTheme |

See : http://doc.prestashop.com/display/PS15/Installing+PrestaShop+using+the+command+line

https://blackfire.io/profiles/compare/347d0b48-8afd-407e-b988-600d086faea3/graph

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
The main interesting information here is "How to test it ?", why do you drop this row ?

**Comment by Shine-neko** [comment]
@mickaelandrieu done ;)

File: classes/shop/Shop.php [hunk 1]
```diff
@@ -135,7 +138,9 @@ public function __construct($id = null, $id_lang = null, $id_shop = null)
 
         if ($this->id) {
             $this->setUrl();
-            $this->setTheme();
+            if ($this->theme == null) {
```
Comment (Quetzacoalt91) [inline classes/shop/Shop.php]: I'm pretty sure this variable always equals null.
This is a attribute of this class, and nothing can alter it before we reach this line.

File: install-dev/models/install.php [hunk 2]
```diff
@@ -910,15 +910,16 @@ public function installFixtures($entity = null, array $data = array())
...
+        $themeName = $themeName ?: _THEME_NAME_;
         $builder = new ThemeManagerBuilder(
             Context::getContext(),
             Db::getInstance()
         );
 
         $theme_manager = $builder->build();
 
-        return $theme_manager->install(_THEME_NAME_) && $theme_manager->enable(_THEME_NAME_);
+        return $theme_manager->install($themeName) && $theme_manager->enable($themeName);
```
Comment (Quetzacoalt91) [inline install-dev/models/install.php]:
```php
return $theme_manager->enable($themeName);
```
would be enough, right ? The install() is needed when we have a zip to extract. (ping @julienbourdeau)

Comment (mickaelandrieu) [inline install-dev/models/install.php]: Agree :)

**Pull request merged**